pkg/manager/member/executor_scaler.go

```diff
@@ -1 +1,186 @@
 package member
+
+import (
+	"context"
+	"fmt"
+	"github.com/pingcap/tiflow-operator/api/v1alpha1"
+	"github.com/pingcap/tiflow-operator/pkg/manager/member/prune"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscaling "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+type executorScaler struct {
+	ClientSet kubernetes.Interface
+	PVCPruner prune.PVCPruner
+}
+
+// NewExecutorScaler return a executorScaler
+func NewExecutorScaler(clientSet kubernetes.Interface) Scaler {
+
+	return &executorScaler{
+		ClientSet: clientSet,
+		PVCPruner: nil,
+	}
+}
+
+func (s *executorScaler) Scale(meta metav1.Object, oldSts *appsv1.StatefulSet, newSts *appsv1.StatefulSet) error {
+
+	actual := *oldSts.Spec.Replicas
+	desired := *newSts.Spec.Replicas
+
+	klog.Infof("start scaling logic, actual: %d, desired: %d",
+		actual, desired)
+
+	scaling := desired - actual
+	if scaling > 0 {
+		return s.ScaleOut(meta, oldSts, newSts)
+	} else if scaling < 0 {
+		return s.ScaleIn(meta, oldSts, newSts)
+	}
+
+	return nil
+}
+
+func (s *executorScaler) ScaleOut(meta metav1.Object, actual *appsv1.StatefulSet, desired *appsv1.StatefulSet) error {
+	tc, ok := meta.(*v1alpha1.TiflowCluster)
+	if !ok {
+		return nil
+	}
+
+	s.PVCPruner = prune.NewPersistentVolumePruner(s.ClientSet, tc)
+	ctx := context.TODO()
+	if !s.PVCPruner.IsStateful() {
+		klog.Info("PVC pruning for Scaling Up")
+		if err := s.PVCPruner.Prune(ctx); err != nil {
+			return err
+		}
+	} else {
+		klog.Info("Scaler will not delete the PVC. Because Executor is stateful")
+	}
+
+	ns := tc.GetNamespace()
+	tcName := tc.GetName()
+	stsName := actual.GetName()
+
+	klog.Infof("start to scaling up tiflow-executor statefulSet %s for [%s/%s]",
+		stsName, ns, tcName)
+
+	up := *desired.Spec.Replicas - *actual.Spec.Replicas
+	current := *actual.Spec.Replicas
+
+	for i := up; i > 0; i-- {
+		klog.Infof("scaling up statefulSet %s, current: %d, desired: %d",
+			stsName, current, current+1)
+
+		if err := s.SetReplicas(ctx, actual, uint(current+1)); err != nil {
+			return err
+		}
+
+		if err := s.WaitUntilRunning(ctx); err != nil {
+			return err
+		}
+
+		if err := s.WaitUntilHealthy(ctx, uint(current+1)); err != nil {
+			return err
+		}
+
+		current++
+		time.Sleep(10 * time.Second)
+	}
+
+	klog.Infof("scaling up is done, tiflow-executor statefulSet %s for [%s/%s]",
+		stsName, ns, tcName)
+
+	klog.Infof("scaling up statefulSet %s, current: %d, desired: %d",
```

Comment:
How about combining these two logs?

Reply:
I thought it would be too long to combine a log here

Reply:
I thought it would be too long to combine a log here